### PR TITLE
Added documentation for plugins

### PR DESCRIPTION
**Pull request by wattson12**
This adds 2 new doc files:
- `Plugins.md` for an overview of plugins and descriptions of the existing core plugins
- `CustomPlugin.md` for an example of creating a new plugin

Original discussion is here: https://github.com/Moya/Moya/issues/397

I wrote this locally and I wasn't sure how to setup links, use swift syntax etc in markdown so any feedback would be great. 

Thanks

### CODE REVIEW

**Comment by ashfurrow** [comment]
Awesome! Looks good to me, want to add links to the new files from the readmes in the docs/ and docs/Examples directories? Good to merge after that 👍

**Comment by colinta** [comment]
@wattson12 I added some links and syntax to your markdown: <https://github.com/wattson12/Moya/pull/1>

File: docs/Plugins.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Plugins
+=======
+
+Moya plugins are used to perform some side effect whenever a request is sent or received. Plugins are defined by the `PluginType` protocol, which can receive callbacks when a request is going to be sent, and when a response has been received. 
+
+##Built in plugins##
+Moya ships with some default plugins which can be used for common functions: authentication, network activity indicator management, and logging.
+
+####Authentication####
```
Comment (ashfurrow) [inline docs/Plugins.md]: Might make sense to remove the trailing `###` here, to be consistent with the other docs?

**Comment by wattson12** [comment]
Thanks @colinta I've merged those changes

@ashfurrow I've just pushed those suggested changes

Thanks both for the feedback / help

**Comment by ashfurrow** [comment]
Fab!

**Pull request merged**